### PR TITLE
AC Partner V3: Add socket support (Closes #337)

### LINKED ISSUE #337: Control Air Conditioner Companion like Xiaomi Mi Smart WiFi Socket
Hi
Is this library can control the Air Conditioner Companion like Xiaomi Mi Smart WiFi Socket ? Just like the MiJia app do, control the output of 220v of the Air Conditioner Companion.

### PR DESCRIPTION
**Pull request by syssi**

### CODE REVIEW

**Comment by syssi** [comment]
I will improve the test coverage a bit tomorrow.

File: miio/airconditioningcompanion.py [hunk 1]
```diff
@@ -208,7 +227,7 @@ def fan_speed(self) -> Optional[FanSpeed]:
     def mode(self) -> Optional[OperationMode]:
         """Current operation mode."""
         try:
-            mode = int(self.data[1][3:4])
+            mode = int(self.data['model_and_state'][1][3:4])
```
Comment (rytilahti) [inline miio/airconditioningcompanion.py]: Considering all of these access the same structure, would it make sense to have a variable for it? Like `self.state = self.data['model_and_state'][1]`?

File: miio/airconditioningcompanion.py [hunk 2]
```diff
@@ -250,7 +269,17 @@ def __json__(self):
...
+
+    def __init__(self, ip: str = None, token: str = None, start_id: int = 0,
+                 debug: int = 0, lazy_discover: bool = True,
+                 model: str = MODEL_ACPARTNER_V2) -> None:
+        super().__init__(ip, token, start_id, debug, lazy_discover)
+
+        if model in MODELS_SUPPORTED:
+            self.model = model
+        else:
+            self.model = MODEL_ACPARTNER_V2
```
Comment (rytilahti) [inline miio/airconditioningcompanion.py]: This defaults to v2 if an unknown model is given? Maybe add a warning logging for that case.

File: miio/airconditioningcompanion.py (hunk 1 above)
Comment (syssi) [reply miio/airconditioningcompanion.py]: Good point!

**Pull request merged**